traffic_ops/install/bin/postinstall.py

```diff
@@ -1167,34 +1338,41 @@ def main(
 			logging.critical("psql is not installed, please install it to continue with database setup")
 			return 1
 
-		try:
-			setup_database_data(conn_str, admin_conf, paramconf, root_dir)
-		except (OSError, subprocess.SubprocessError)as e:
+		def db_connect_failed():
 			logging.error("Failed to set up database: %s", e)
 			logging.error(
-				"Can't connect to the database.  " \
-				"Use the script `/opt/traffic_ops/install/bin/todb_bootstrap.sh` " \
+				"Can't connect to the database.  "
+				"Use the script `/opt/traffic_ops/install/bin/todb_bootstrap.sh` "
 				"on the db server to create it and run `postinstall` again."
 			)
+
+		try:
+			setup_database_data(conn_str, admin_conf, paramconf, root_dir)
+		except (OSError, subprocess.CalledProcessError)as e:
+			db_connect_failed()
+			return -1
+		except subprocess.SubprocessError as e:
```

Comment:
Why split these `excepts` if they have the same handler?

Reply:
`subprocess.SubprocessError` is not defined in Python 2. Separating them like that prevent
```
AttributeError: 'module' object has no attribute 'SubprocessError'
```
in Python 2

Reply:
really? I would expect that to still give an error. I suppose if the test passes that means it's fine.